### ISSUE TITLE
MACD returning different result from tradingview's MACD

### ISSUE THREAD

**Issue by mmaestrini** [ixjb94/indicators#9]
Hi again,

I have been trying to get the MACD working like on my tradingview but to no avail.

Tradingview's macd shows last 4 histograms values as -3, 1.3, 0.9, 3 vs what I get is -0.3164, -0.25768, -0.24459, -0.13909.

I configured my macd in the library as `ta.macd(close1m, 12, 26, 9)` and  passing the close bars of the btc 1min klinedata and I belive the configuration is as it should according to my tradingview setup 
![image](https://github.com/ixjb94/indicators/assets/66750632/2d1b940a-4dc5-4eb2-b716-8443c1b73445)

Can you perhaps tell me what I am doing wrong?

**Comment by ixjb94** [comment]
@mmaestrini Hi.
Thanks for the report.
Here is more precise MACD calculation.  Update to the latest version.
https://github.com/ixjb94/indicators/commit/2ab4f90e42f635c9196513c636fa305a8396f071

**Comment by VRciF** [comment]
Hi @ixjb94,
I was just experimenting with a few libraries and noticed, that macd differs greatly from the original `talib` c implementation.  
For comparision, a nodejs implementation, which uses talib and computes the same values as binance, can be found here: https://github.com/oransel/node-talib  
I'm not sure about the difference and if the values wrongly calculated or maybe you follow another definition of macd.  
Maybe give node-talib a try and compare if the values actually differ  

As an example I used the values of ADAUSDT from 2024-10-02 00:00:00 to 2024-10-03 23:59:59 with 30m candles, see
https://api.binance.com/api/v3/klines?symbol=ADAUSDT&interval=30m&startTime=1727820000000&endTime=1727992799000  

The close price has index 4 - see the docs for an example and details: https://binance-docs.github.io/apidocs/spot/en/#kline-candlestick-data  

Also, the php implementation https://github.com/LupeCode/phpTraderNative/blob/master/source/TALib/Core/MomentumIndicators.php#L841  
also computes values equal to binance and talib

**Comment by VRciF** [comment]
Sorry, for highjacking this old thread.  
I'm not sure anymore if there is an issue.  

I mean, the values differ for sure for a given dataset, but regarding binance, I found this [thread](https://github.com/LupeCode/phpTraderNative/issues/11).  
The key parts are:
* Maybe, binance uses another EMA smoothing factor than the standard `2`, but instead uses `20`. Not sure if this is true
* And the values seem to better align if more historic candles are loaded beforehand. This is also what I have found myself. In the thread it is mentioned that with 1500 candles one should be on the safe side and be pretty accurate compared to binance

This kind of makes sense, as the EMA formula requires the previous EMA value. Thus if there are any differences, this might get worse. And I'm not sure about the convergence properties of EMA in general.
